tests/common.h

```diff
@@ -48,6 +51,28 @@ void random_shuffle (RandomAccessIterator first, RandomAccessIterator last)
 
 void waitForSignal(QObject *object, const char *signal);
 
+class ContactChangeListener : public SeasideCache::ChangeListener
+{
+public:
+    ContactChangeListener();
+    ~ContactChangeListener();
+
+    bool waitForContactAdded(int contactId);
+    bool waitForContactDeleted(int contactId);
+
+private:
+    bool waitForChange(quint32 contactId, QSet<quint32> *changeWaits, QSet<quint32> *changeRecords);
+    void itemUpdated(SeasideCache::CacheItem *item);
+    void itemAboutToBeRemoved(SeasideCache::CacheItem *item);
+
+    QSet<quint32> m_updatedContactIds;
+    QSet<quint32> m_deletedContactIds;
+    QSet<quint32> m_waitForContactAddedId;
+    QSet<quint32> m_waitForContactDeletedId;
+    QEventLoop *m_loop = nullptr;
```

Comment:
OOI why are these heap-allocated (pointers) rather than being stack allocated members?

Reply:
No particular reason, either way would be okay.